### DISCUSSION TITLE
<title>Building a Error message

### DISCUSSION THREAD

**Discussion by C11H17N3O-8** [community/community#40426] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Below is criteria 

main()
This function shall use the command line arguments to read the primary and secondary sensor values from the specified files and then continue to prepare the dictionaries so that they meet the requirements needed to call the function that finds the faulty sensors. Finally the function shall find and print the faulty sensors. The main function shall use the other functions to accomplish its tasks and it shall have error handling that prints the message below if the files given as command line arguments are not valid:

"Error: The files given as arguments are not valid."

I know that the building of an error is should be made in the main function but I do not understand why my code is not to be accepted. Below is my code in python

import pickle
import sys

a1 = {'kitchen': '23*', 'bedroom':'22*', 'bathroom':'24*', 'garage':'28*'}
a2 = {'kitchen': '26*', 'bedroom':'22*', 'bathroom':'24*', 'garage':'28*'}

with open('filename1.pickle', 'wb') as handle:
    pickle.dump(a1, handle, protocol=pickle.HIGHEST_PROTOCOL)

with open('filename2.pickle', 'wb') as handle:
    pickle.dump(a2, handle, protocol=pickle.HIGHEST_PROTOCOL)

def read_file(filename):
    with open(filename, 'rb') as file:
        loaded_dictionary = pickle.load(file)
    return loaded_dictionary

def map_to_int(measurements):
    converted_dictionary = {}
    for key,value in measurements.items():
        value = value[:-1] 
        value = int(value) 
        converted_dictionary[key] = value 
    return converted_dictionary

def find_faulty(primary, secondary, threshold):
    faulty_set = set()
    for key,value in primary.items():
        if  abs(primary[key] - secondary[key]) > threshold:
            faulty_set.add(key)
    return faulty_set

def display_warnings(faulty_sensors):
    print("\nAnalyzis of the provided files is complete.\n\
            -------------------------------------------\
            \nThe following sensors differ more than 2°:\n")
    
    for item in faulty_sensors:
        print(item)

   
def main():
    try:
        arg1 = sys.argv[1]
        arg2 = sys.argv[2]
    except FileNotFoundError as e:
        raise e ("Usage: {sys.argv[0]} <filename1_path> <filename2_path>")
    
    dictionary1 = read_file(arg1)
    dictionary2 = read_file(arg2)
    
    dictionary1 = map_to_int(dictionary1)
    dictionary2 = map_to_int(dictionary2)
    
    faulty_sensors = find_faulty(dictionary1,dictionary2,2)
    display_warnings(faulty_sensors)

if __name__ == "__main__":
    main()

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @C11H17N3O-8! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
